mmv1/api/type.go

```diff
@@ -270,7 +275,47 @@ type Type struct {
 
 const MAX_NAME = 20
 
-// func (t *Type) validate() {
+func (t *Type) Validate(r *Resource) {
```

Comment:
This isn't really a validation function anymore is it? It looks like it really just sets default values. What do you think?

Reply:
Yes, here it sets default values. We can have two functions to separate setting default values and validations (if the field is required or not).